common/src/main/java/com/microsoft/identity/common/internal/cache/MsalCppOAuth2TokenCache.java

```diff
@@ -123,19 +127,73 @@ public synchronized void saveCredentials(@Nullable final AccountRecord accountRe
      *
      * @param accountRecord : accountRecord to be saved.
      */
-    public synchronized void saveAccountRecord(@NonNull AccountRecord accountRecord) {
+    public synchronized void saveAccountRecord(@NonNull final AccountRecord accountRecord) {
         getAccountCredentialCache().saveAccount(accountRecord);
-
     }
 
     /**
      * API to clear all cache.
      * Note: This method is intended to be only used for testing purposes.
      */
+    @VisibleForTesting(otherwise = VisibleForTesting.NONE)
     public synchronized void clearCache() {
         getAccountCredentialCache().clearAll();
     }
 
+    /**
+     * API to inspect cache contents.
+     * Note: This method is intended to be only used for testing purposes.
+     *
+     * @return A immutable List of Credentials contained in this cache.
+     */
+    @VisibleForTesting(otherwise = VisibleForTesting.NONE)
+    public synchronized List<Credential> getCredentials() {
+        return Collections.unmodifiableList(
+                getAccountCredentialCache().getCredentials()
+        );
+    }
+
+    /**
+     * Force remove an AccountRecord matching the supplied criteria.
+     *
+     * @param homeAccountId HomeAccountId of the Account.
+     * @param environment   The Environment of the Account.
+     * @param realm         The Realm of the Account.
+     * @return An {@link AccountDeletionRecord} containing a receipt of the removed Accounts.
+     * @throws ClientException
+     */
+    @VisibleForTesting // private by default for production code
+    public synchronized AccountDeletionRecord forceRemoveAccount(@NonNull final String homeAccountId,
+                                                                 @Nullable final String environment,
+                                                                 @NonNull final String realm) throws ClientException {
+        validateNonNull(homeAccountId, "homeAccountId");
+        validateNonNull(realm, "realm");
+
+        final boolean mustMatchOnEnvironment = !StringExtensions.isNullOrBlank(environment);
+
+        final List<AccountRecord> removedAccounts = new ArrayList<>();
+
+        for (final AccountRecord accountRecord : getAllAccounts()) {
```

Comment:
Don't quite remember how JAVA executes this....would the `getAllAccounts` method be executed on each iteration of the loop? Would it be better to call this method before the loop and store the result in a variable and pass that to the loop?

Reply:
This syntax is shorthand for using an iterator; `getAllAccounts()` will be executed once, then an iterator from it will be returned.  Example below:
![image](https://user-images.githubusercontent.com/990063/84929031-21f38300-b084-11ea-89c5-3bbb665756f7.png)